frontends/p4/sideEffects.cpp

```diff
@@ -237,6 +273,7 @@ class DismantleExpression : public Transform {
             auto block = new IR::BlockStatement(*ifFalse);
             auto ifStatement = new IR::IfStatement(expression->srcInfo, cond, ifTrue, block);
             result->statements->push_back(ifStatement);
+            // TODO: result->addAssignment() returns a clone; why clone again?
```

Comment:
Also, better safe than sorry. Unless the contract for addAssignment is clear about this, I don't want to rely on it.